### DISCUSSION TITLE
Toggle options like `--no-multi-line` or `--no-hscroll`

### DISCUSSION THREAD

**Discussion by amariusz** [junegunn/fzf#4135] [Q&A] [answered]
I wonder if it's possible to change (or even better to toggle)  `--no-multi-line` or `--no-hscroll` without restarting fzf. I'm using `--read0` to obtain multi-line list and sometimes, depending on query results, it would be convenient to toggle multiline on and off (along with `no-hscroll`).

I've seen some examples using `transform` but there are no `actions` related to these command line options.

Is there any other way?

**Comment by junegunn** [comment]
No, there currently isn't a way. But adding it would be trivial. What is the exact use case?

**Comment by amariusz** [comment]
Thank you for replying! That's good to know that internal logic is not preventing such behaviour after fzf is executed :)

My usecase is similar to your example:
```
declare -f |
  perl -0 -pe 's/^}\n/}\n\0/gm' |
  bat --plain --language bash --color always |
  fzf --read0 --ansi --layout reverse --multi --highlight-line
```
The length of my input entries (basically text documents) varies greatly - from 5 to 1000 lines. Most of them are short and browsing with fzf list is enough. But depending on the query I may end up with those bulky ones. They don't fit on the screen, occupies whole list and most of the time the match is not visible anyway. In such cases I prefer to 'fold' the document on the list to singe line with `--no-multi-line`, prevent it from scrolling with `--no-hscroll` (because first line has some crucial metadata like creation date),  and use preview to browse the document. 

Currently I need to restart fzf with different options for such cases but using  shortcut(s) to toggle them when needed would more comfortable.

**Comment by amariusz** [comment]
Something related I was thinking about - would it be also trivial to toggle `--read0` option this way?

The use case would be 2-stage filtering, similar to your example here: https://junegunn.github.io/fzf/tips/ripgrep-integration/
First stage is filtering whole 'documents' with `--read0` - for example only those that have (or don't have) specific #hashtag. Then switching modes with something like `ctrl-t` which also turns off `--read0` and filtering lines within those documents with separate query.

This is already possible with another very useful approach - select all matching entries ('documents'),  execute vim with {+f} and use BLines to perform second stage filtering.

**Reply by junegunn** [reply]
No. It's very different from `toggle-multi-line` which only affects the user interface. `--read0` is used to control how fzf parses the input stream. Once read, changing the option afterwards won't affect the items already read. To change the list, you have to use `reload`.

**Reply by amariusz** [reply]
You're right, this is important distiction.
I was actually thinking about using `reload`. Sorry for linking wrong example - I've realized there are few examples of rg integration in your documentation. This is the one I had in mind: https://github.com/junegunn/fzf/blob/master/ADVANCED.md#switching-between-ripgrep-mode-and-fzf-mode

I guess this could probably work if instead of rg, another fzf process with  `--read0` `--print0` and `--filter` would be used to do the first stage filtering (whole documents) and then after pressing ctrl-f (and enabling search) line filtering would be in effect.

**Comment by junegunn** [comment, marked as answer]
See ca5e633 and e60a9a6.

**Reply by amariusz** [reply]
Thank you very much! Changes seem OK to me :)
